components/DataBrowser/DatasetFiles.vue

```diff
@@ -0,0 +1,209 @@
+      file_path = file_path + `/${this.$route.query.path.slice(6)}`;
+      this.generateFilesBreadcrumb(this.$route.query.path);
+    }
+    this.updateFilesData(file_path);
+  },
+
+  watch: {
+    '$route.query.path': {
+      handler(new_val, old_val) {
+      let file_path = `/tempZone/home/rods/12L/datasets/${this.$route.params.id}`;
+        if (this.breadcrumb[this.breadcrumb.length - 1].to !== new_val) {
+          if (new_val && new_val !== 'files')
+            file_path = file_path + `/${new_val.slice(new_val.indexOf('/') + 1)}`;
+          this.updateFilesData(file_path, 'Folder', old_val);
+        }
+      }
+    }
+  },
+
+  computed: {
+    title: function() {
+      return this.breadcrumb[this.breadcrumb.length - 1].label;
+    }
+  },
+
+  methods: {
+    generateFilesBreadcrumb(fullpath){
+      let path = 'files';
+      fullpath.split('/').forEach((item) => {
+        if (item !== 'files') {
+          path = path + '/' + item;
+          this.breadcrumb.push({
+            to: path,
+            label: item
+          })
+        }
+      });
+    },
+
+    handleFilesData(data) {
+      data.folders.forEach((item) => {
+        let folder = {
+          name: item.name,
+          path: item.path,
+          type: "Folder"
+        }
+        this.files_data.push(folder);
+      })
+      data.files.forEach((item) => {
+        let file = {
+          name: item.name,
+          path: item.path,
+          type: "File"
+        }
+        this.files_data.push(file);
+      })
+    },
+
+    async updateFilesData(path, type = 'Folder', old_path = this.$route.query.path) {
+      if (type === "Folder") {
+        this.isLoadingFile = true;
+        let folder_path = 'files';
+        if (path.split('/').length > 7) {
+          folder_path = folder_path + `/${path.slice(path.indexOf(this.$route.params.id) + this.$route.params.id.length + 1)}`;
+        }
+        if (old_path && folder_path.length < old_path.length) {
+          this.breadcrumb = this.breadcrumb.filter((item) => {
+            if (folder_path.indexOf(item.label) > -1)
+              return item;
+          })
+        } else if (old_path && folder_path.length > old_path.length) {
+          this.breadcrumb.push({
+            to: folder_path,
+            label: path.slice(path.lastIndexOf('/') + 1)
+          })
+        }
+        this.$router.push({
+          path: `${this.$route.path}`,
+          query: {
+            datasetTab: `${this.$route.query.datasetTab}`,
+            path: folder_path
+          }
+        });
+        this.files_data = [];
+        let new_files = await backendQuery.fetchFiles(this.get_file_path, {path: path});
+        this.handleFilesData(new_files);
+        this.isLoadingFile = false;
+      } else {
+        this.downloadFile(path);
+      }
+    },
+
+    handlePath(path) {
+      let file_path = `/tempZone/home/rods/12L/datasets/${this.$route.params.id}`;
```

Comment:
Is it possible to repalce the hardcoded prefix /tempZone/home/rods/12L/datasets ?